server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java

```diff
@@ -45,6 +51,25 @@ public class HealthPeriodicLogger implements ClusterStateListener, Closeable, Sc
     public static final String HEALTH_FIELD_PREFIX = "elasticsearch.health";
     public static final String MESSAGE_FIELD = "message";
 
+    // Writers for logs or messages
+    // default visibility for testing purposes
+    BiConsumer<LongGaugeMetric, Long> metricWriter = LongGaugeMetric::set;
+    Consumer<ESLogMessage> logWriter = logger::info;
```

Comment:
Rather than keep these as mutable class variables, I think they should be `final`, and then passed in to the constructor (the `create` method). Then the tests can use the visible constructor to pass in custom lambda methods for testing.